.github/workflows/build.yml

```
@@ -0,0 +1,29 @@
+name: Build and Test
+
+on:
+  push:
+    branches: [ master ]
+    tags-ignore:
+    - 'v*'
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-18.04
```

```suggestion
    strategy:
      fail-fast: false
      matrix:
        arch: [x64, x86]
        os: [ubuntu-18.04, ubuntu-latest]
        compiler: [gcc, clang]
    runs-on: ${{ matrix.os }}
```

I was just browsing through and thought I'd suggest some type of matrix... though not necessarily this one. :)

---

Point taken. I'm focusing on the release machinery in this PR but more extensive matrix testing is a good thing. I'd like the matrix test workflow to run early in the integration pipeline so the release workflow can stay streamlined. I'm having to repeat a lot of build code between workflows, though. Ready to learn.

---

>  I'm having to repeat a lot of build code between workflows, though. Ready to learn.

I'm not too great with Actions yet... your yml looks pretty good though, a lot more advanced than I have done.

Of course one way to prevent code repetition in these workflow files is using shell scripts, but I didn't see where you could do that in the use-cases you need.

One thing that might help trim a few lines is to combine some of the name/run lines. You don't need a "name" for every "run". Example:

```
      - run: ./autogen
      - run: ./configure
      - run: make
      - run: make check
      - run: make distcheck
```

or 

```
      - run:  |
         ./autogen
         ./configure
         make
         make check
         make distcheck
```

Or because that sequence of commands is used in build.yml and milestones.yml, put them in a shell script.

```
- run: sh .github/scripts/build.sh
```

---

Oh, you could probably actually remove the `make check` since `make distcheck` does that automatically. 

Also, I find that using `make V=1` can be handy sometimes